redis-gin-mongo/src/analytics_worker/main.go

```
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+var MONGO2_HOST = os.Getenv("MONGO2_HOST")
+var MONGO2_PORT = os.Getenv("MONGO2_PORT")
+var REDIS_HOST = os.Getenv("REDIS_HOST")
+var REDIS_PORT = os.Getenv("REDIS_PORT")
+
+type AnalyticsData struct {
+	Title string
+	Views int
+}
+
+func getDoc(title string, mongo2 *mongo.Client) AnalyticsData {
+	coll := mongo2.Database("blog").Collection("views")
+	var result AnalyticsData
+	err := coll.FindOne(context.TODO(), bson.D{{"title", title}}).Decode(&result)
+	if err != nil {
+		fmt.Println(err)
```

return the error in this case and return 404:
```


if err == mongo.ErrNoDocuments {
// This error means your query did not match any documents.
return NewViewsNotFoundErr()
}


```

(you should return an error that is not mongo specific to allow other databases to return the same error and not change the rest of the code).